common/txmgr/inmemory_store.go

```diff
@@ -231,7 +233,37 @@ func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Updat
 }
 
 func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) SaveFetchedReceipts(ctx context.Context, receipts []R, chainID CHAIN_ID) error {
-	return nil
+	if ms.chainID.String() != chainID.String() {
+		return fmt.Errorf("save_fetched_receipts: %w: %s", ErrInvalidChainID, chainID)
+	}
+
+	// Persist to persistent storage
+	if err := ms.persistentTxStore.SaveFetchedReceipts(ctx, receipts, chainID); err != nil {
+		return err
+	}
+
+	// Update in memory store
+	var errs error
+	ms.addressStatesLock.RLock()
+	defer ms.addressStatesLock.RUnlock()
+	for _, receipt := range receipts {
+		var found bool
+		var receiptErrs error
+		for _, as := range ms.addressStates {
+			err := as.moveTxToConfirmed(receipt)
+			if err == nil {
+				found = true
+				break
+			}
+			receiptErrs = errors.Join(receiptErrs, err)
+		}
+		if !found {
+			receiptErrs = fmt.Errorf("receipt was not found: %w", receiptErrs)
```

Comment:
Technically we didn't find the tx for the given receipt. Could you update this error message to reflect?